pages/docs/cache.en-US.md

```diff
@@ -0,0 +1,94 @@
+# Custom Cache
+
+By default, SWR is using a global cache to store data sharing among all hooks. Besides that, there's a new way to customize your own cache provider.
+New property `cache` and a new API `createCache` are now introduced in `swr@beta`. This API is intended to resolve problem of using SWR with various, diverse storages.
+
+## Create Custom Cache
+
+### `createCache`
+
+This API receive a underlay cache `provider` as argument. Returns an object, with `cache` instance that could be consumed by SWR hooks,
+and `mutate` API to manipulate the corresponding cache. Note that it's not the global `mutate` API.
+
+```js
+const { mutate, cache } = createCache(provider)
+```
+
+You can pass down `cache` through SWRConfig or the `useSWR` hook options.
+
+```jsx
+import { SWRConfig, createCache } from 'swr'
+
+const provider = new Map()
+
+const { mutate, cache } = createCache(provider)
+
+// pass to SWR context
+<SWRConfig value={{ cache }}>
+  <Page />
+</SWRConfig>
+
+// or pass to hook options
+useSWR(key, fetcher, { cache })
+```
+
+### `provider`
+
+The provider is used to let user manage cache values directly, and the interface should match the following definition:
+
+```ts
+interface Cache<Data = any> {
+  get(key: Key): Data | null | undefined
+  set(key: Key, value: Data): void
+  delete(key: Key): void
+}
+```
+
+Those methods are being used inside SWR to manage cache. Beyond SWR itself, now user can access the cached keys, values from `provider` directly.
+For instance if the provider is a Map instance, you'll be able to access the used keys through provider by using `Map.prototype.keys()`.
+
```

Comment:
(TODO: add note that mentions that you shouldn't **write** to cache, you should always use `mutate`.)